geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HdelDUnitTest.java

```diff
@@ -0,0 +1,165 @@
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.dunit.rules.RedisClusterStartupRule;
+import org.apache.geode.test.junit.rules.ExecutorServiceRule;
+
+public class HdelDUnitTest {
+
+  @ClassRule
+  public static RedisClusterStartupRule cluster = new RedisClusterStartupRule();
+
+  @ClassRule
+  public static ExecutorServiceRule executor = new ExecutorServiceRule();
+
+  private static final int HASH_SIZE = 50000;
+  private static MemberVM locator;
+  private static MemberVM server1;
+  private static MemberVM server2;
+  private static int[] redisPorts;
+  private static RedisCommands<String, String> lettuce;
+  private static StatefulRedisConnection<String, String> connection;
+  private static ClientResources resources;
+
+  @BeforeClass
+  public static void classSetup() {
+    redisPorts = AvailablePortHelper.getRandomAvailableTCPPorts(3);
+
+    String redisPort1 = "" + redisPorts[0];
+    String redisPort2 = "" + redisPorts[1];
+
+    locator = cluster.startLocatorVM(0);
+
+    server1 = startRedisVM(1, redisPorts[0]);
+    server2 = startRedisVM(2, redisPorts[1]);
+
+    DUnitSocketAddressResolver dnsResolver =
+        new DUnitSocketAddressResolver(new String[] {redisPort2, redisPort1});
+
+    resources = ClientResources.builder()
+        .socketAddressResolver(dnsResolver)
+        .build();
+
+    RedisClient redisClient = RedisClient.create(resources, "redis://localhost");
+    redisClient.setOptions(ClientOptions.builder()
+        .autoReconnect(true)
+        .build());
+    connection = redisClient.connect();
+    lettuce = connection.sync();
+  }
+
+  private static MemberVM startRedisVM(int vmID, int redisPort) {
+    int locatorPort = locator.getPort();
+
+    return cluster.startRedisVM(vmID, x -> x
+        .withConnectionToLocator(locatorPort)
+        .withProperty(REDIS_PORT, "" + redisPort));
+  }
+
+  @Before
+  public void testSetup() {
+    lettuce.flushall();
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    resources.shutdown().get();
+    connection.close();
+    server1.stop();
+    server2.stop();
+  }
+
+  @Test
+  public void testConcurrentHDelReturnExceptedNumberOfDeletions() {
+    AtomicLong client1Deletes = new AtomicLong();
+    AtomicLong client2Deletes = new AtomicLong();
+
+    String key = "HSET";
+
+    Map<String, String> setUpData =
+        makeHashMap(HASH_SIZE, "field", "value");
+
+    lettuce.hset(key, setUpData);
+
+    new ConcurrentLoopingThreads(HASH_SIZE,
+        i -> {
+          long deleted = lettuce.hdel(key, "field" + i, "value" + i);
+          client1Deletes.addAndGet(deleted);
+        },
+        i -> {
+          long deleted = lettuce.hdel(key, "field" + i, "value" + i);
+          client2Deletes.addAndGet(deleted);
+        })
+            .run();
+
+    assertThat(client1Deletes.get() + client2Deletes.get()).isEqualTo(HASH_SIZE);
+  }
+
+  @Test
+  public void testConcurrentDel_whenServerCrashesAndRestarts() {
```

Comment:
Since you have to re-trigger anyway, you could change the name to HDel instead of Del and better fit with the test name above: `testConcurrentHDel_whenServerCrashesAndRestarts_deletesAllHashFieldsAndValues`

Reply:
Done